pulse/interface/viewer_3d/render_widgets/psd_preview_render_widget.py

```diff
@@ -0,0 +1,146 @@
+                tube.SetRadius(section_data[0] / 2)
+                tube.SetNumberOfSides(50)
+                tube.CappingOn()
+                tube.Update()
+
+                filter.AddInputData(tube.GetOutput())
+                filter.Update()
+            
+            if segment_label == connection_point:
+                sphere = vtkSphereSource()
+                sphere.SetCenter(*start_coords)
+                sphere.SetRadius(section_data[0] / 4)
+                sphere.Update()
+
+                sphere_mapper = vtkPolyDataMapper()
+                sphere_mapper.SetInputData(sphere.GetOutput())
+
+                sphere_actor = vtkActor()
+                sphere_actor.SetMapper(sphere_mapper)
+
+                sphere_actor.GetProperty().SetColor(1, 0, 0) 
+
+                self.add_actors(sphere_actor)
+
+
+    
+        mapper = vtkPolyDataMapper()
+        filter_actor = vtkActor()
+
+        mapper.SetInputData(filter.GetOutput())
+        filter_actor.SetMapper(mapper)
+        filter_actor.GetProperty().SetInterpolationToPhong()
+        filter_actor.GetProperty().SetDiffuse(0.8)
+        filter_actor.GetProperty().SetSpecular(1.5)
+        filter_actor.GetProperty().SetSpecularPower(80)
+        filter_actor.GetProperty().SetSpecularColor(1, 1, 1)
+
+        self.add_actors(filter_actor)
+
+    def config_view(self):
+        camera = self.renderer.GetActiveCamera()
+        camera.SetPosition(1, 1, 1)
+        camera.SetFocalPoint(0, 0, 0) 
+        self.renderer.ResetCamera()
+        self.renderer.ResetCameraClippingRange()        
+        self.renderer.ResetCameraScreenSpace()
+            
+    def update_theme(self):
+        user_preferences = app().main_window.config.user_preferences
+        bkg_1 = user_preferences.renderer_background_color_1
+        bkg_2 = user_preferences.renderer_background_color_2
+        font_color = user_preferences.renderer_font_color
+
+        if bkg_1 is None:
+            raise ValueError('Missing value "bkg_1"')
+        if bkg_2 is None:
+            raise ValueError('Missing value "bkg_2"')
+        if font_color is None:
+            raise ValueError('Missing value "font_color"')
+
+        self.renderer.GradientBackgroundOn()
+        self.renderer.SetBackground(bkg_1.to_rgb_f())
+        self.renderer.SetBackground2(bkg_2.to_rgb_f())
+
+        if hasattr(self, "text_actor"):
+            self.text_actor.GetTextProperty().SetColor(font_color.to_rgb_f())
+
+        if hasattr(self, "colorbar_actor"):
+            self.colorbar_actor.GetTitleTextProperty().SetColor(font_color.to_rgb_f())
+            self.colorbar_actor.GetLabelTextProperty().SetColor(font_color.to_rgb_f())
+
+        if hasattr(self, "scale_bar_actor"):
+            self.scale_bar_actor.GetLegendTitleProperty().SetColor(font_color.to_rgb_f())
+            self.scale_bar_actor.GetLegendLabelProperty().SetColor(font_color.to_rgb_f())
+
+
+    def set_theme(self, *args, **kwargs):
+        self.update_theme()
```

Comment:
you could have implemented the "update_theme" function directly inside "set_theme"